server/app/models/User.js

```diff
@@ -0,0 +1,63 @@
+const mongoose = require('mongoose');
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+
+const options = {
+  timestamps: true,
+};
+
+const UserSchema = mongoose.Schema(
+  {
+    name: {
+      type: String,
+      required: [true, 'Please add a name'],
+    },
+    email: {
+      type: String,
+      required: [true, 'Please add an email'],
+      unique: true,
+      match: [
+        // eslint-disable-next-line no-useless-escape
+        /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/,
+        'Please add a valid email',
+      ],
+    },
+    password: {
+      type: String,
+      required: [true, 'Please add a password'],
+      minlength: 6,
+      select: false,
+    },
+  },
+  options,
+);
+
```

Comment:
I think it would be a good idea to have [custom error handling](https://mongoosejs.com/docs/middleware.html#error-handling-middleware) here so that you can pass user-friendly messages to the frontend

E.g "The email entered is already taken" instead of "duplicated field value entered"